docs/book/src/component-config-tutorial/testdata/project/tools.go

```diff
@@ -0,0 +1,25 @@
+//go:build tools
+
+/*
+Copyright 2022 The Kubernetes authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// +kubebuilder:docs-gen:collapse=Apache License
+
+// This package imports things required by build scripts, to force go mod to see them as dependencies
+package tools
+
+import (
+	_ "sigs.k8s.io/controller-runtime/tools/setup-envtest"
+)
```

Comment:
This setup is going to result in the project's CI dependencies being added to the main project's go.mod file, which has implications on:
- how underlying dependencies get resolved when a build tool and the main project share a transitive dependency (e.g. the build tool might unnecessarily hold back the transitive dependency on a lower version due to Go's MVS algorithm)
- how other projects consume the main project's as a dependency (external projects will end up with this project's build tool dependencies in their go.mod files and might impact their transitive dependency selection also due to MVS)

TL;DR: At a minimum, I think we need to put the tools dependencies in a separate go module. However see also this issue for background about why even that might not be a great idea: https://github.com/kubernetes-sigs/cluster-api/issues/5569

/hold